### PR TITLE
Add/getAllUsers

### PR DESCRIPTION
**Pull request by anotheredward**

### CODE REVIEW

File: ldap.production.js [hunk 1]
```diff
@@ -43,3 +46,13 @@ function getUserGroups(username, callback) {
       }))
   })
 }
+
+function getAllUsers(callback) {
+  ad.getUsersForGroup({ attributes: ['cn', 'sAMAccountName'] }, allUserGroup, remapUsers)
+
+  function remapUsers(err, data) {
+    callback(err, data.map(function (user) {
```
Comment (anotheredward) [inline ldap.production.js:50]: ad.getUsersForGroup returns an array by default, so I haven't included a guard.

File: ldap.development.js [hunk 2]
```diff
@@ -4,10 +4,24 @@ module.exports = function (config) {
...
 function mockGetUserGroups(username, callback) {
   callback(null, [userGroups[username]])
 }
+
+function mockGetAllUsers(callback) {
+  callback(null, [
+      {
+        id: 'jtaylor',
+        name: 'James Taylor'
+      },
```
Comment (markstuart) [inline ldap.development.js:21]: I feel like there should be another mock user here:

```
{
  id: 'taytay',
  name: 'Taylor Swift'
},
```

**Comment by markstuart** [comment]
Looks good :+1:

File: ldap.development.js (hunk 2 above)
Comment (anotheredward) [inline ldap.development.js:21]: Fixed.

**Pull request merged**